packages/vc-handler-ld-local/src/ld-document-loader.ts

```diff
@@ -0,0 +1,113 @@
+        const resolutionResult = await context.agent.resolveDid({ didUrl: url })
+
+        const didDoc = resolutionResult.didDocument
+        if (!didDoc) {
+          throw new Error(`Could not fetch DID document with url: ${url}. Did you enable the the driver?`)
+        }
+
+        // Move legacy publicKey to verificationMethod, so any dependency that does not support it, keeps functioning
+        if (didDoc.publicKey) {
+          if (!didDoc.verificationMethod) {
+            didDoc.verificationMethod = []
+          }
+          didDoc.verificationMethod = [...didDoc.verificationMethod, ...didDoc.publicKey]
+          delete didDoc.publicKey
+        }
+
+        if (url.indexOf('#') > 0 && didDoc['@context']) {
+          // Apparently we got a whole DID document, but we are looking for a verification method
+          const component = await context.agent.getDIDComponentById({ didDocument: didDoc, didUrl: url })
+          if (component && component.id) {
+            // We have to provide a context
+            const contexts = this.ldSuiteLoader
+              .getAllSignatureSuites()
+              .filter((x) => x.getSupportedVerificationType() === component.type /* || component.type === 'Ed25519VerificationKey2018'*/)
+              .filter((value, index, self) => self.indexOf(value) === index)
+              .map((value) => value.getContext())
+            const fragment = { ...component, '@context': contexts }
+
+            return {
+              contextUrl: null,
+              documentUrl: url,
+              document: fragment,
+            }
+          }
+        }
+
+        // currently Veramo LD suites can modify the resolution response for DIDs from
+        // the document Loader. This allows to fix incompatibilities between DID Documents
+        // and LD suites to be fixed specifically within the Veramo LD Suites definition
+        this.ldSuiteLoader.getAllSignatureSuites().forEach((x) => x.preDidResolutionModification(url, didDoc))
+
+        // console.log(`Returning from Documentloader: ${JSON.stringify(returnDocument)}`)
+        return {
+          contextUrl: null,
+          documentUrl: url,
+          document: didDoc,
+        }
+      }
+
+      if (this.ldContextLoader.has(url)) {
+        const contextDoc = await this.ldContextLoader.get(url)
+        return {
+          contextUrl: null,
+          documentUrl: url,
+          document: contextDoc,
+        }
+      } else {
+        if (attemptToFetchContexts) {
+          debug('WARNING: attempting to fetch the doc directly for ', url)
+          try {
+            const response = await fetch(url)
+            if (response.status === 200) {
+              const document = await response.json()
+              // fixme: The VC API returns an _id object for RL Credentials, which is not allowed, so delete it here for now
+              if (url.startsWith('https://vc-api.sphereon.io/services/credentials/')) {
```

Comment:
Can it be fixed in this ticket?

Reply:
No, as it is an external issue with the VC API. I did not create a ticket for it, since the VC API will be completely rewritten using this SDK anyway